### PR TITLE
pythonPackages.ete3: init at 3.1.2

### PR DESCRIPTION
**Pull request by delehef**
###### Motivation for this change

The ETE toolkit is a reference python library for manipulation of phylogenetic trees in Newick or Phyloxml format.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? **No**
- [x] Tested **package tests are broken; tested with `pythonImportsCheck`**
- [x] Tested compilation of all packages that depend on this change **no packages depend on this change**
- [x] Tested basic functionality of all **library files**
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ete3/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  pythonImportsCheck = [ "ete3" ];
+
+  propagatedBuildInputs = [ six numpy ]
+                          ++ lib.optional withTreeVisualization (if isPy3k then pyqt5 else pyqt4)
+                          ++ lib.optional withXmlSupport lxml;
+
+  meta = with lib; {
+    description = "A Python framework for the analysis and visualization of trees";
+    homepage = "http://etetoolkit.org/";
+    license = licenses.gpl3;
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/ete3/default.nix:27]: `gpl3Only` or `gpl3Plus`?

File: pkgs/development/python-modules/ete3/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+  doCheck = false; # Tests are (i) not 3.x compatible, (ii) broken under 2.7
+  pythonImportsCheck = [ "ete3" ];
+
+  propagatedBuildInputs = [ six numpy ]
+                          ++ lib.optional withTreeVisualization (if isPy3k then pyqt5 else pyqt4)
+                          ++ lib.optional withXmlSupport lxml;
+
+  meta = with lib; {
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/ete3/default.nix:24]: `meta.maintainers` seems to be missing

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Also, could you run `nixpkgs-fmt` on this?

**Comment by delehef** [comment]
Thanks for your comments; I fixed them in the last commits.

File: pkgs/development/python-modules/ete3/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+, lxml
+, withXmlSupport ? false
+, pyqt4
+, pyqt5
+}:
+
+buildPythonPackage rec {
+  pname = "ete3";
+  version = "3.1.2";
+  src = fetchPypi {
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/ete3/default.nix:17]:
```suggestion
  version = "3.1.2";
  
  src = fetchPypi {
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Just one more nit, suprised `nixpkgs-fmt` didn't catch it.

Also, could you squash your commits, so that you only have two? Something like
```commit
maintainers: add delehef
python3Packages.ete3: init at 3.1.2
```
would be perfect :)

**Comment by delehef** [comment]
Should be alright now!

**Comment by IvarWithoutBones** [comment]
When running nixpkgs-review on this, i got a missing module error when trying to execute `bin/ete3`: `ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'`. Nix-code wise though, this looks good to me :relaxed:

**Comment by delehef** [comment]
Indeed, I was able to reproduce the error. Turns out that `nixpkgs-review rev master` setup two instances of ete3 (`/nix/store/XXX-python3.10-ete3-3.1.2/bin/ete3` & `/nix/store/YYY-python3.9-ete3-3.1.2/bin/ete3`) for Python 3.9 & 3.10, and that the highest one in `$PATH` is the one for Python 3.10.

However, for some reason, when importing numpy in Python 3.10, it calls upon numpy for 3.9, which you can observe by running `import numpy` in `python3.10` in the very same environment. Python 3.9 still imports its own numpy though, and thus does not meet the same problem.

Finally, calling directly ete for Python 3.9 (`/nix/store/YYY-python3.9-ete3-3.1.2/bin/ete3` or `/nix/store/ZZZ/bin/ete3
`) fails with a `cannot import name 'TextFace' from 'ete3'` – which is expected as I didn't enable Qt5 support by default.

I'm not sure if this problem is stemming from my side or from `nixpkgs-review`, as I confess I arrive at the very fringe of my Nix competences.

**Comment by delehef** [comment]
Bump?

**Comment by delehef** [comment]
Bump again?

**Comment by onny** [comment]
Result of `nixpkgs-review pr 150972` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.ete3</li>
    <li>python39Packages.ete3</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

**Pull request merged**